bin/prettier.js

```diff
@@ -67,25 +67,52 @@ if (argv["help"] || (!filepatterns.length && !stdin)) {
   process.exit(argv["help"] ? 0 : 1);
 }
 
-function getParser() {
+function getParserOption() {
+  const optionName = "parser";
+  const value = argv[optionName];
+
+  if (value === undefined) {
+    return value;
+  }
+
   // For backward compatibility. Deprecated in 0.0.10
   if (argv["flow-parser"]) {
     console.warn("`--flow-parser` is deprecated. Use `--parser flow` instead.");
     return "flow";
   }
 
-  if (argv["parser"] === "flow") {
-    return "flow";
+  if (value === "flow" || value === "babylon") {
+    return value;
   }
 
+  console.warn(
+    "Ignoring unknown --" + optionName + ' value, falling back to "babylon":\n' +
+    '  Expected "flow" or "babylon", but received: ' + JSON.stringify(value)
+  );
+
   return "babylon";
 }
 
+function getIntOption(optionName) {
+  const value = argv[optionName];
+
+  if (value === undefined) {
+    return value;
+  }
+
+  if (/^\d+$/.test(value)) {
+    return Number(value);
+  }
+
```

Comment:
All our int options right now are assumed to be positive integers. It crashes in random ways if you put negative numbers. May want to enforce it here